Physical_Engine/Query/GeometricalCentreline.cs

```diff
@@ -0,0 +1,78 @@
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Geometry;
+using BH.Engine.Spatial;
+using BH.oM.Geometry;
+using BH.oM.Physical.Elements;
+using BH.oM.Physical.FramingProperties;
+using BH.oM.Reflection.Attributes;
+using System.ComponentModel;
+
+namespace BH.Engine.Physical
+{
+    public static partial class Query
+    {
+        [Description("Returns the centreline of an IFramingElement's section bounding box.")]
+        [Input("element", "The IFramingElement to query the geometrical centreline of.")]
+        [Output("curve", "The geometrical centreline of the IFramingElement.")]
+        public static ICurve GeometricalCentreline(this IFramingElement element)
+        {   
+            ICurve location = element?.Location;
+            if (location == null)
+                return null;
+
+            Vector normal = null;
+            normal = element.Normal();
+            if (normal == null)
+            {
+                Engine.Reflection.Compute.RecordError("IFramingElement must have linear location line.");
+                return null;
+            }
+
+            Vector tangent = null;
+            tangent = location.ITangentAtPoint(location.ICentroid());
+            if (tangent == null)
+                return null;
+
+            Vector thirdAxis = tangent.CrossProduct(normal);
+
+            if (element.Property is ConstantFramingProperty)
+            {
+                Point geometricalCentre = (element.Property as ConstantFramingProperty)?.Profile?.Edges?.Bounds()?.Centre();
+
+                if (geometricalCentre == null)
+                    return null;
+
+                Vector sectionTranslate = geometricalCentre - Point.Origin;
+
+                return location.ITranslate(tangent * sectionTranslate.X + thirdAxis * sectionTranslate.Y + normal * sectionTranslate.Z);
```

Comment:
I think `normal * sectionTranslate.Z` term is not needed here - we do not want to move the curve along the element.
```suggestion
                return location.ITranslate(tangent * sectionTranslate.X + thirdAxis * sectionTranslate.Y);
```